back/transcendence_django/shared_models/models.py

```diff
@@ -77,4 +96,33 @@ def __str__(self):
 
     async def save_game_summary(self, game_summary: GameSummary) -> None:
         await sync_to_async(self.game_summaries.add)(game_summary)
+        self.__update_time_played(game_summary)
+        self.__update_count(game_summary)
+        if game_summary.is_remote:
+            self.__update_win_dict(game_summary)
         await sync_to_async(self.save)()
+
+    def __update_win_dict(self, game_summary) -> None:
+        if str(game_summary.winner_user_id) == str(self.id):
+            key = WIN
+        elif game_summary.winner_user_id is None:
+            key = TIE
+        else:
+            key = LOSS
+        self.win_loss_tie[key] += 1
+
+    def __update_time_played(self, game_summary) -> None:
+        start_time = game_summary.start_time or datetime.min
+        end_time = game_summary.end_time or datetime.max
+        game_duration: int = int((end_time - start_time).total_seconds())
+        if game_summary.is_remote:
+            self.time_played[REMOTE] += game_duration
+        else:
+            self.time_played[LOCAL] += game_duration
+
+    def __update_count(self, game_summary) -> None:
+        if game_summary.is_remote:
+            self.game_counter[REMOTE] += 1
+        else:
+            self.game_counter[LOCAL] += 1
+        self.game_counter[TOTAL] += 1
```

Comment:
Consistency w.r.t. win dict and time played: either have a total value for all, or for none? Additions are fun.